api/src/main/java/org/pmiops/workbench/cdr/CdrDbConfig.java

```diff
@@ -77,11 +85,22 @@ public void CdrDataSource(CdrVersionDao cdrVersionDao) {
       afterPropertiesSet();
     }
 
+    @PostConstruct
+    public void init() {
+      finishedInitialization = true;
+    }
+
     @Override
     protected Object determineCurrentLookupKey() {
       CdrVersion cdrVersion = CdrVersionContext.getCdrVersion();
       if (cdrVersion == null) {
-        throw new ServerErrorException("No CDR version specified!");
+        if (finishedInitialization) {
+          throw new ServerErrorException("No CDR version specified!");
+        } else {
+          // If Spring is still initializing, return the the default CDR version
```

Comment:
I don't really follow, can you be more specific? What state are we in where Spring hasn't initialized, yet we're running in an autowired method and we have a handle to a working `defaultCdrVersionProvider`?

Reply:
Updated the comment. Let me know if that makes more sense.